precalculus/exercises/outcomes/FN/FN5/generator.sage

```diff
@@ -1,10 +1,40 @@
 class Generator(BaseGenerator):
     def data(self):
+
+      fname1,fname2,fname3,fname4 = sample(["f","g","h","p","q","r"],4)
+
+      functions = [choice([2..6])*x+choice([-5..5]),
+                            choice([2..4])*x^2+choice([-3..3])*x^(choice([0,1])),
+                            choice([1,2])*x^3+choice([-3..3])*x^(choice([0,1,2])),
+                            sqrt(choice([2..5])*x+choice([-4..4])),
+                            choice([1..4])*x/(x+choice([-5..-1,1..5]))
+                            ]
+      #f3(x)=choice(functions)
+      f1,f2,f3,f4 = sample(functions,4)
+      swap=choice([True,False])
+      xval=choice([-5..-1,1..5])
+      if swap:
+        comp = f4.subs(x=f3)
+        compx= f3.subs(x=f4).subs(x=xval)
+      else:
+        comp = f3.subs(x=f4)
+        compx= f4.subs(x=f3).subs(x=xval)
+
       return {
-        "eq": 3,
-        "ineq": 4,
-        "eqansinterval": 5,
-        "eqansgraph": 6,
-        "ineqansinterval": 7,
-        "ineqansgraph": 8
+        "funct1": f1,
+        "funct2": f2,
+        "funct3": f3,
+        "funct4": f4,
+        "fname1": fname1,
+        "fname2": fname2,
+        "fname3": fname3,
+        "fname4": fname4,
+        "sum": f1+f2,
+        "difference": f1-f2,
+        "product": f1*f2,
```

Comment:
```suggestion
        "product": expand(f1*f2),
```

Again maybe - the sum and difference are simplified so it's not clear to me the point here.

Of course we could make the denominator function always be monomial, so the quotient can always be simplified (modulo zero-denominator situations I guess).

Reply:
Specific request in #168:
>Answers not completely multiplied out and simplified if possible. Have them still look similar to the outside functions